src/patternfly/layouts/Toolbar/styles.scss

```diff
@@ -0,0 +1,146 @@
+@import "../../patternfly-utilities";
+
+// The value of component scoped variables is always defined by a global variable:
+.pf-l-toolbar {
+
+  /* stylelint-disable */
+
+  // Toolbar Section
+  --pf-l-toolbar__section--MarginTop:           var(--pf-global--spacer--md);
+  --pf-l-toolbar__section--PaddingTop:          var(--pf-global--spacer--md);
+  --pf-l-toolbar__section--PaddingBottom:       var(--pf-global--spacer--md);
+  --pf-l-toolbar__section--BorderColor:         var(--pf-global--BorderColor);
+  --pf-l-toolbar__section--BorderTopWidth:      var(--pf-global--BorderWidth--sm);
+  --pf-l-toolbar__section--BorderBottomWidth:   var(--pf-global--BorderWidth--sm);
+
+  // Toolbar Group
+  --pf-l-toolbar__group--MarginRight: var(--pf-global--spacer--xl);
+  --pf-l-toolbar__group--MarginLeft:  var(--pf-global--spacer--xl);
+
+  // Toolbar Item
+  --pf-l-toolbar__item--MarginRight:  var(--pf-global--spacer--md);
+  --pf-l-toolbar__item--MarginLeft:   var(--pf-global--spacer--md);;
+
+  /* stylelint-enable */
+
+  // toolbar, toolbar__group, toolbar__section shared styles
+  &,
+  &__section,
+  &__group {
+    display: flex;
+    flex-wrap: wrap;
+    align-items: center;
+  }
+
+  // Section
+  &__section {
+    flex-basis: 100%;
+
+    // Add margin-top for stacked sections
+    &:not(:first-child) {
+      margin-top: var(--pf-l-toolbar__section--MarginTop);
+    }
+  }
+
+  // Group
+  &__group:not(:last-child) {
+    margin-right: var(--pf-l-toolbar__group--MarginRight);
+  }
+
+  // Item
+  &__item {
+    // Setting line-height in order to vertically align properly. Ex: badge
+    line-height: 1;
+
+    &:not(:last-child) {
+      margin-right: var(--pf-l-toolbar__item--MarginRight);
+    }
+  }
+
+  // Modifiers
+  // Set flex items to nowrap
+  &.pf-m-nowrap,
+  .pf-m-nowrap {
+    flex-wrap: nowrap;
+  }
+
+  // Set flex items to align top
```

Comment:
Do we need these modifiers? I think that by default the tool bar items should align to the center.

If we do start adding this kind of modifiers maybe is a good case to do the `pf-flex` layout.

Reply:
Agreed